x/feemarket/keeper/params.go

```diff
@@ -58,24 +56,28 @@ func (k Keeper) GetBaseFeeEnabled(ctx sdk.Context) bool {
 }
 
 // GetBaseFee gets the base fee from the store
-func (k Keeper) GetBaseFee(ctx sdk.Context) *big.Int {
+func (k Keeper) GetBaseFee(ctx sdk.Context) math.LegacyDec {
 	params := k.GetParams(ctx)
 	if params.NoBaseFee {
-		return nil
+		return math.LegacyDec{}
 	}
 
-	baseFee := params.BaseFee.BigInt()
-	if baseFee == nil || baseFee.Sign() == 0 {
+	baseFee := params.BaseFee
+	if baseFee.IsNil() || baseFee.IsZero() {
+		bfV1 := k.GetBaseFeeV1(ctx)
+		if bfV1 == nil {
+			return math.LegacyDec{}
```

Comment:
We enter the if when baseFee is Nill or zero, but if also BaseFeeV1 is nil we always return nil, not the value of baseFee. Is this correct?

Reply:
It's correct because technically is not a nil pointer, is calling LegacyDec.IsNil() that is asking if it's an empty LegacyDec struct (the object exists but with no value set)